### ISSUE TITLE
Dropbox uses a different hash algorithm than digest

### ISSUE THREAD

**Issue by richfitz** [karthik/rdrop2#157]
(this is not an rdrop2 bug but a suggestion for an enhancement)

Dropbox use an odd way of hashing files - https://www.dropbox.com/developers/reference/content-hash

This means that the `content_hash` returned by `rdrop2::drop_dir()` do not match those computed by running `sha256` on the file (or `digest::digest(readBin(path, raw(), file.size(path)), algo = "sha256", serialize = FALSE)`)

The solution is fairly straightforward, and I have a function for computing this hash.  Is this something you would like added to the package?  If so I can file a PR.  (perhaps `drop_content_hash` as the function name?)

Please start your issue with a brief description of the problem.

Then, fill out as much as you can below, excluding parts you're not sure of or which don't apply. We'd rather know about an issue than not, so don't worry if you can't answer everything.

The rest of this applies functional issues (code); remove it for other problems, like issues about documentation.

Provide code to reproduce the issue. This will usually need to involve code to create and upload one or more files/folders. Below is an example you can edit.

```r
library(rdrop2)

drop_auth()

# create & upload some file
path <- "mtcars.csv"
write.csv(mtcars, path)
drop_upload(path)

# the problem call
info <- drop_dir()
info$content_hash[info$name == path] 
## [1] 8c00dcec5f3e6bf58a42dcf354f0d5199a43567e88a9d80291bd2b85f53a54a5
digest::digest(readBin(path, raw(), file.size(path)), algo = "sha256", serialize = FALSE)
## [1] 439ba335c3d28dd0c1871f75bdffb389d5a3b23cf703275566700140c9523ae5
```

## Expected Behaviour

The hashes would be the same

## Actual Behaviour

The hashes are different

## Proposed fix

A function that can be called like

```r
drop_content_hash(path)
[1] "8c00dcec5f3e6bf58a42dcf354f0d5199a43567e88a9d80291bd2b85f53a54a5"
```

## Account Type

Both basic and business

## Session Info

Replace the output below (inside the backticks) with output from `devtools::session_info()` (or `sessionInfo()` if you don't have `devtools` installed)

<details><pre>

```r
Session info ------------------------------------------------------------------
 setting  value                       
 version  R version 3.3.3 (2017-03-06)
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_GB.UTF-8                 
 tz       GB-Eire                     
 date     2018-01-26                  

Packages ----------------------------------------------------------------------
 package              * version date       source                         
 assertive              0.3-5   2016-12-31 CRAN (R 3.3.3)                 
 assertive.base         0.0-7   2016-12-30 CRAN (R 3.3.3)                 
 assertive.code         0.0-1   2015-10-06 CRAN (R 3.3.3)                 
 assertive.data         0.0-1   2015-10-06 CRAN (R 3.3.3)                 
 assertive.data.uk      0.0-1   2015-10-06 CRAN (R 3.3.3)                 
 assertive.data.us      0.0-1   2015-10-06 CRAN (R 3.3.3)                 
 assertive.datetimes    0.0-2   2016-05-10 CRAN (R 3.3.3)                 
 assertive.files        0.0-2   2016-05-10 CRAN (R 3.3.3)                 
 assertive.matrices     0.0-1   2015-10-06 CRAN (R 3.3.3)                 
 assertive.models       0.0-1   2015-10-06 CRAN (R 3.3.3)                 
 assertive.numbers      0.0-2   2016-05-09 CRAN (R 3.3.3)                 
 assertive.properties   0.0-4   2016-12-30 CRAN (R 3.3.3)                 
 assertive.reflection   0.0-4   2016-12-30 CRAN (R 3.3.3)                 
 assertive.sets         0.0-3   2016-12-30 CRAN (R 3.3.3)                 
 assertive.strings      0.0-3   2016-05-10 CRAN (R 3.3.3)                 
 assertive.types        0.0-3   2016-12-30 CRAN (R 3.3.3)                 
 assertthat             0.2.0   2017-04-11 CRAN (R 3.3.3)                 
 base                 * 3.3.3   2017-03-06 local                          
 codetools              0.2-15  2016-10-05 CRAN (R 3.3.1)                 
 curl                   3.0     2017-10-06 cran (@3.0)                    
 datasets             * 3.3.3   2017-03-06 local                          
 DBI                    0.7     2017-06-18 CRAN (R 3.3.3)                 
 devtools               1.13.4  2017-11-09 CRAN (R 3.3.3)                 
 digest                 0.6.12  2017-01-27 CRAN (R 3.3.2)                 
 dplyr                  0.5.0   2016-06-24 CRAN (R 3.3.1)                 
 graphics             * 3.3.3   2017-03-06 local                          
 grDevices            * 3.3.3   2017-03-06 local                          
 httpuv                 1.3.5   2017-07-04 cran (@1.3.5)                  
 httr                   1.3.1   2017-08-20 cran (@1.3.1)                  
 jsonlite               1.5     2017-06-01 cran (@1.5)                    
 knitr                  1.17    2017-08-10 CRAN (R 3.3.3)                 
 magrittr               1.5     2014-11-22 CRAN (R 3.2.2)                 
 memoise                1.1.0   2017-08-10 Github (hadley/memoise@d63ae9c)
 methods              * 3.3.3   2017-03-06 local                          
 openssl                0.9.9   2017-11-10 CRAN (R 3.3.3)                 
 purrr                  0.2.2   2016-06-18 CRAN (R 3.3.1)                 
 R6                     2.2.2   2017-06-17 cran (@2.2.2)                  
 Rcpp                   0.12.12 2017-07-15 cran (@0.12.12)                
 rdrop2               * 0.8.1   2017-09-29 CRAN (R 3.3.3)                 
 stats                * 3.3.3   2017-03-06 local                          
 stringi                1.1.3   2017-03-21 CRAN (R 3.3.3)                 
 tibble                 1.2     2016-08-26 CRAN (R 3.3.1)                 
 tools                  3.3.3   2017-03-06 local                          
 utils                * 3.3.3   2017-03-06 local                          
 withr                  2.0.0   2017-07-28 cran (@2.0.0)                  
```

</pre></details>

**Comment by karthik** [comment]
> Is this something you would like added to the package? If so I can file a PR. (perhaps drop_content_hash as the function name?)


Yes Rich, that would be fantastic!

**Comment by richfitz** [comment]
Submitted as https://github.com/karthik/rdrop2/pull/158

**Comment by ClaytonJY** [comment]
Closed by #158.
